### PR TITLE
Update validate content logic to correctly handle missing keys 

### PR DESCRIPTION
**Pull request by pschoffer**
## Changelog Description

Validate content tries to build a Elastic Search document from post and compare it with the one actually stored in Elastic Search.
Before this change, the code would ignore any keys missing in either indexed or the newly build documen. This change makes these differences to be acknowledged and potentially used to trigger auto-healing process.  

An example of this is a new meta key being added to be indexed. After the change, we will recognize that there is a difference for all documents that should have but are missing that meta. And we will slowly reindex them as part of validate-content logic even without explicit re-index call.  

### Plugin Updated: Search

This change upgrades validate contents health check to also trigger on missing keys.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1.  (one of way to get inindexed key) register custom taxanomy:
```
register_taxonomy(
	'collections',
	['post', 'page'],
	[
		'show_admin_column' => true,
		'hierarchical' => true,
		'show_in_rest' => true,
		'public' => false,
		'sort' => false,
		'show_ui' => true,
	]
);
```
1. Add this taxanomy to a page/post in wp-admin
2. register taxanomy to be indexed (important to do AFTER you added the taxanomy)
```
add_filter(
	'vip_search_post_taxonomies_allow_list',
	function( $taxonomies ) {
		$taxonomies[] = 'collections';
		return $taxonomies;
	}
);
```
3. validate contents dev-env exec -- wp vip-search health validate-contents --start_post_id=5 --last_post_id=5

### CODE REVIEW

**Review by pauarge — APPROVED** [review]
LGTM

**Comment by WPprodigy** [comment]
Think we need to make similar adjustments to `simplified_diff_document_and_prepared_document()` as well.

File: search/includes/classes/class-health.php [hunk 1]
```diff
@@ -595,14 +595,27 @@ public static function filter_expected_post_rows( $rows, $post_types, $post_stat
...
+		$checked_keys = [];
+
 		foreach ( $document as $key => $value ) {
+			$checked_keys[ $key ] = true;
 			if ( in_array( $key, self::DOCUMENT_IGNORED_KEYS, true ) ) {
 				continue;
 			}
 
 			if ( is_array( $value ) ) {
-				$recursive_diff = self::simplified_diff_document_and_prepared_document( $document[ $key ], $prepared_document[ $key ] );
```
Comment (pschoffer) [inline search/includes/classes/class-health.php:604]: Cha turns out the simple one used to be very shallow too :) The `$recursive_diff` was ignored before.

**Comment by rinatkhaziev** [comment]
@pschoffer can we add a little bit more context in the title and description, please? I.e. the "why" and the "how".

**Review by rinatkhaziev — APPROVED** [review]
👍

**Pull request merged**